contracts/OpenOraclePriceData.sol

```diff
@@ -43,9 +41,11 @@ contract OpenOraclePriceData is OpenOracleData {
 
         // Only update if newer than stored, according to source
         Datum storage prior = data[source][key];
-        if (prior.timestamp < timestamp) {
+        if (prior.timestamp < timestamp && timestamp < block.timestamp && block.timestamp - 10 minutes < timestamp) {
```

Comment:
check the message was signed in the last 10 minutes. totally random guess at the value we want, up for discussion

Reply:
This might be a little stringent, I have mixed feelings about relying too much on block.timestamp since it slightly creates a new risk vector. What about just timestamp < block.timestamp + 10 minutes?

Reply:
yeah, a buffer for future timestamps makes sense. you dont think we should prevent posting stale prices? what about if the limit was 1 hour

Reply:
I'm trying to think why we should ever reject a price that's fresher than the one on-chain

Reply:
i actually kind of agree.. @hayesgm ?